dockstore-webservice/src/main/java/io/dockstore/webservice/core/Collection.java

```diff
@@ -52,13 +52,14 @@
 @ApiModel("Collection")
 @Schema(name = "Collection", description = "Collection in an organization, collects entries")
 @Entity
-@Table(name = "collection")
+@Table(name = "collection", uniqueConstraints = @UniqueConstraint(name = "unique_collection_display_name", columnNames = {"organizationid", "displayname"}))
 @NamedQueries({
         @NamedQuery(name = "io.dockstore.webservice.core.Collection.getByAlias", query = "SELECT e from Collection e JOIN e.aliases a WHERE KEY(a) IN :alias"),
         @NamedQuery(name = "io.dockstore.webservice.core.Collection.findAllByOrg", query = "SELECT col FROM Collection col WHERE organizationid = :organizationId"),
         @NamedQuery(name = "io.dockstore.webservice.core.Collection.deleteByOrgId", query = "DELETE Collection c WHERE c.organization.id = :organizationId"),
         @NamedQuery(name = "io.dockstore.webservice.core.Collection.findAllByOrgId", query = "SELECT c from Collection c WHERE c.organization.id = :organizationId"),
         @NamedQuery(name = "io.dockstore.webservice.core.Collection.findByNameAndOrg", query = "SELECT col FROM Collection col WHERE lower(col.name) = lower(:name) AND organizationid = :organizationId"),
+        @NamedQuery(name = "io.dockstore.webservice.core.Collection.findByDisplayNameAndOrg", query = "SELECT col FROM Collection col WHERE lower(col.displayName) = lower(:displayName) AND organizationid = :organizationId"),
```

Comment:
The DB constraint is case-insensitive, this query is not. Somehow, some way, in a very rare case, this might eventually cause a problem --- somehow Foo and FOO get inserted into the DB (I know you guard against it now, but code moves around), and when you use this query, who knows which one you'll get.

http://shuber.io/case-insensitive-unique-constraints-in-postgres/ -- but you have to create a new data type and add an extenion. May be overkill.

Reply:
Interesting find. 
I don't like the extension idea. I do like the idea at the very bottom of the article to just make an index on the lower-case version of the names. I don't think we ever will want duplicates differing only by case. I wonder if unique constraints being case sensitive will cause  issues elsewhere, 

i.e. doing something like 
`CREATE UNIQUE INDEX email_unique_idx on users (LOWER(displayname)); `

Reply:
Thinking about this more, I'm not sure my use case would actually be that bad.  The Java code wouldn't let you create a duplicate case-insensitive display name through that path, but if it somehow got into the DB (manually, through another code path), I think all our code would work.

But if it's easy to create the index like you suggest, then we probably should, if that's what we want to enforce.